gemp-lotr/gemp-lotr-async/src/main/web/src/updated/jCard/modules/image-resolution/image-resolver.ts

```diff
@@ -0,0 +1,62 @@
+import { getCardImage } from "../../types/card-image.functions";
+import { hobbitCards } from "./data/hobbit.list";
+import { choiceBoosters, fellowshipPacks, gempLotrPromo, holidayOverrides, postMoviePacks, randoms, returnOfTheKingPacks, setStarterPacks, specialStarters, tengwarSelections, twoTowersPacks } from "./data/image-overrides.list";
+import { playerCommitteeCards } from "./data/players-committee";
+import { set40Cards } from "./data/set40.list";
+import { Override } from "./types/override.interface";
+
+/**
+ * NOTE: No additional images should be added in this location.
+ * All updates should be provided in the data folder and bubble up through
+ * the players-committee/index.ts file
+ */
+
+let imageCache: Map<string, string> | null = null;
+
+const buildImageResolver = (): Map<string, string> => {
+    const overrides = new Map<string, string>();
+    
+    addImageLocations(overrides, holidayOverrides);
+    addImageLocations(overrides, gempLotrPromo);
+    addImageLocations(overrides, setStarterPacks);
+    addImageLocations(overrides, tengwarSelections);
+    addImageLocations(overrides, choiceBoosters);
+    addImageLocations(overrides, fellowshipPacks);
+    addImageLocations(overrides, twoTowersPacks);
+    addImageLocations(overrides, returnOfTheKingPacks);
+    addImageLocations(overrides, postMoviePacks);
+    addImageLocations(overrides, specialStarters);
+    addImageLocations(overrides, randoms);
+    addImageLocations(overrides, set40Cards);
+    addImageLocations(overrides, hobbitCards);
+
+    addImageLocations(overrides, playerCommitteeCards);
+
+    return overrides;
+}
+
+const addImageLocations = (overrides: Map<string, string>, newItems: Override[]): void => {
+    newItems.forEach(override => {
+        overrides.set(override.id, override.url)
+    });
+}
+
+const getImageCache = (): Map<string, string> => {
+    if (!imageCache) {
+        imageCache = buildImageResolver();
+    }
+    return imageCache;
+}
+
+export const getImageUrl = (blueprintId: string): string => {
+    const cacheValue = getImageCache().get(blueprintId);
```

Comment:
if item is in cache use it. (O(1) lookup)
Otherwise build the URL.
If you found no image (???) return an empty string. Maybe it should be a constant "no image found" image
if you found image then save that into the cache.

Reply:
i'll make a compromise for you :) We use that 404 page and add a query param that the wiki isn't expecting. Then it gets rendered as the 404 and we can still see the query param in the URL so somebody could even just copy paste it to us. 

Something like: `https://wiki.lotrtcgpc.net/images/Card_Not_Found.jpg?originalImage=01_012`

Reply:
Sounds like a plan